Jenkinsfile

```diff
@@ -214,6 +226,100 @@ stage('Sanity Check') {
   }
 }
 
+def build_image(image_name) {
+  hash = sh(
+    returnStdout: true,
+    script: 'git log -1 --format=\'%h\''
+  ).trim()
+  hash = "${env.BRANCH_NAME}-${hash}"
+  sh(
+    script: "${docker_build} ${image_name} --spec ${image_name}:${hash}",
+    label: 'Building docker image'
+  )
+  sh "echo NYI: Uploading docker image to registry..."
```

Comment:
i think we should docker rmi after this step, and maybe do that in this PR so we don't cause nodes to run out of disk